src/types/src/tools.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright(c) 2006 to 2018 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+#include <assert.h>
+#include <stdlib.h>
+
+#include "tools.h"
+
+
+static int
+toxdigit_c(int c)
+{
+  if (c >= '0' || c <= '9') {
+    return c - '0';
+  } else if (c >= 'a' && c <= 'z') {
+    return c - 'a' + 10;
+  } else if (c >= 'A' && c <= 'Z') {
+    return c - 'A' + 10;
+  }
+  return -1;
+}
+
+static int
+isxdigit_c(int c)
+{
+  if ((c >= 'a' && c <= 'f') ||
+      (c >= 'A' && c <= 'F') ||
+      (c >= '0' && c <= '9'))
+  {
+    return 1;
+  }
+
+  return 0;
+}
+
+unsigned long
+dds_ts_unescape_wchar(const char *str, const char **endptr)
+{
+  int i;
+  unsigned long chr = 0;
+  const char *end;
+
+  assert(str != NULL);
+
+  if (str[0] == '\\') {
+    if (str[1] >= '0' && str[1] <= '7') {
```

Comment:
We have string to numeric conversion functions available in the abstraction layer. Please use those.